R/addins.R

```diff
@@ -0,0 +1,111 @@
+#' Style the active file
+#'
+#' Helper function for RStudio Addin.
+style_active_file <- function() {
+  file <- rstudioapi::getActiveDocumentContext()$path
+  style_file(file, style = tidyverse_style)
+}
+
+
+#' Style the highlighted region
+#'
+#' Helper function for RStudio Addin. This function is complicated because of
+#' one thing: You can highlight also just parts of lines.
+#' @importFrom rlang seq2
+style_active_region <- function() {
+  context <- find_active_context()
+  all_text <- utf8::read_lines_enc(context$path)
+  styled_expr <- style_region(all_text, context)
+  neighbourhood <- extract_neighbourhood(all_text, context)
+
+  styled_lines <- complete_styled_expr(context, styled_expr, neighbourhood)
+
+  merged_text <- append(neighbourhood$body, styled_lines, context$start[1] - 1)
+  utf8::write_lines_enc(merged_text, context$path)
+}
+
+#' Helper to extract and preprocess relevant attributes from
+#' [rstudioapi::getActiveDocumentContext()].
+#'
+find_active_context <- function() {
+  context <- rstudioapi::getActiveDocumentContext()
```

Comment:
I thought we could just create a simple forwarder to `rstudioapi::getActiveDocumentContext()` in our package, and mock this forwarder:

```r
get_rstudio_context <- function() {
  rstudioapi::getActiveDocumentContext()
}
```

mockr still works if functions are called indirectly from our package (it patches all functions in our package).

Reply:
Ok yes I see. That's what I a say unsure about (see comment above).